clang/lib/3C/3C.cpp

```diff
@@ -96,6 +96,31 @@ ArgumentsAdjuster addVerifyAdjuster() {
     return AdjustedArgs;
   };
 }
+ArgumentsAdjuster canonicalizeSourceFilePathAdjuster() {
+  return [](const CommandLineArguments &Args, StringRef FileName) {
+    std::string AbsoluteFp;
+    std::error_code EC = tryGetCanonicalFilePath(std::string(FileName), AbsoluteFp);
+    assert(!EC && "We canonicalized the source file path during 3C "
+                  "initialization; doing it again in "
+                  "canonicalizeSourceFilePathAdjuster shouldn't fail.");
+    CommandLineArguments AdjustedArgs;
+    for (auto Arg : Args) {
+      if (Arg == FileName) {
+        // XXX We are assuming that any occurrence of the source file path as an
```

Comment:
For the DV options (and any others available at the same time) you can copy `clang/lib/tooling/Tooling.cpp:ASTBuilderAction` (~20 lines) and run it with tool.run() below instead of tool.buildasts(). That's how we currently get our ASTs, and was likely the model for how we did in the past with the actionfactory thing.

That may take a while and this solution is fine for what it is.

Reply:
> For the DV options (and any others available at the same time) you can copy `clang/lib/tooling/Tooling.cpp:ASTBuilderAction` (~20 lines) and run it with tool.run() below instead of tool.buildasts(). That's how we currently get our ASTs, and was likely the model for how we did in the past with the actionfactory thing.

I was wondering if that might work.  Agreed, let's look into migrating as many argument adjusters as possible to this new design later.

Reply:
I filed #536 and updated the comment in the code to reference it.